api/v1alpha2/virtualmachine_network_types.go

```diff
@@ -178,6 +178,27 @@ type VirtualMachineNetworkSpec struct {
 	// +optional
 	Disabled bool `json:"disabled,omitempty"`
 
+	// Nameservers is a list of IP4 and/or IP6 addresses used as DNS
+	// nameservers. These are applied globally.
+	//
+	// Please note this feature is available only with the following bootstrap
+	// providers: LinuxPrep and Sysprep.
+	//
+	// Please note that Linux allows only three nameservers
+	// (https://linux.die.net/man/5/resolv.conf).
+	//
+	// +optional
+	Nameservers []string `json:"nameservers,omitempty"`
+
+	// SearchDomains is a list of search domains used when resolving IP
+	// addresses with DNS. These are applied globally.
+	//
+	// Please note this feature is available only with the following bootstrap
+	// providers: LinuxPrep and Sysprep.
+	//
+	// +optional
+	SearchDomains []string `json:"searchDomains,omitempty"`
```

Comment:
I think we want to depart from the pattern here and state:

```go
// Please note global search domains are only available with the following
// bootstrap providers: LinuxPrep and Sysprep. The Cloud-Init bootstrap
// provider supports per-interface search domains.
```

Otherwise I am concerned users will not realize search domains are supported at all with Cloud-Init.